pkg/kubelet/kubelet.go

```diff
@@ -1556,11 +1561,43 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	}
 	switch internalFieldPath {
 	case "status.podIP":
-		return pod.Status.PodIP, nil
+		return kl.getPodIP(pod)
 	}
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+// getPodIP lazily finds the pod IP by consulting the following sources in order:
+//
+// 1.  The status of the passed in pod on the pod itself
```

Comment:
Naive question: Why can't kubelet wait until it can get access to Pod IP from a single source of truth? I assume the source of truth is the status manager as of now.